docs/src/components/theme-switcher.tsx

```diff
@@ -0,0 +1,55 @@
+import { MoonIcon, SunIcon } from "@radix-ui/react-icons";
+
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+
+const setTheme = (theme: "light" | "dark" | "system") => {
+  const de = document.documentElement;
+
+  de.classList.remove("dark");
+
+  if (theme === "system") {
+    const { matches: prefersDark } = window.matchMedia(
+      "(prefers-color-scheme: dark)",
+    );
+
+    if (prefersDark) {
+      de.classList.add("dark");
+    } 
+
+    localStorage.removeItem("theme");
+  } else {
+    de.classList.add(theme);
```

Comment:
You'll add the "light" class here in case the user selects "light" as a theme and never remove it, even if the user selects "dark" later on. This wouldn't have any impact, but it's funky. Let's only add the "dark" class.

 ```suggestion
    if (theme === "dark") {
      de.classList.add("dark");
    }
```